pkg/option/config.go

```diff
@@ -3337,8 +3349,27 @@ func InitConfig(programName, configName string) func() {
 			} else {
 				// replace deprecated fields with new fields
 				ReplaceDeprecatedFields(m)
-				err := MergeConfig(m)
-				if err != nil {
+				var confArgs []string
+				for key := range m {
+					if key == "cluster-id" {
+						fmt.Println("cluster-id")
+					}
+					value := fmt.Sprintf("%v", m[key])
+					// The config flags may differ from the CLI flags
+					if CheckExistingFlagCLI(key) {
```

Comment:
What's a key that is not a CLI flag? Is that possible?

Reply:
I have met several such flags:

- operator-api-serve-addr
- custom-cni-conf
- cilium-endpoint-gc-interval
- wait-bpf-mount

Reply:
@aanm Is that expected? :thinking:

Reply:
@pchaigno yes, they can be used in the daemonset itself. For example:
```yaml
        - name: CILIUM_WAIT_BPF_MOUNT
          valueFrom:
            configMapKeyRef:
              key: wait-bpf-mount
              name: cilium-config
              optional: true
```